internal/testrunner/runners/system/runner.go

```diff
@@ -1417,16 +1431,56 @@ func (r *runner) validateTestScenario(ctx context.Context, result *testrunner.Re
 		return result.WithError(err)
 	}
 
-	// TODO: remove this once Elasticsearch will map event.original correctly (https://github.com/elastic/elasticsearch/pull/106714)
+	atLeast814, err := semver.NewConstraint(">=8.14")
+	if err != nil {
+		// Pre 8.14 Elasticsearch commonly has event.original not mapped correctly, exclude from check: https://github.com/elastic/elasticsearch/pull/106714
+		return nil, fmt.Errorf("failed to parse version constraint: %w", err)
+	}
+	skipIgnoredFields := append([]string(nil), config.SkipIgnoredFields...)
+	isAtLeast814 := atLeast814.Check(semver.MustParse(r.stackVersion.Number))
+	if !isAtLeast814 {
+		skipIgnoredFields = append(skipIgnoredFields, "event.original")
+	}
```

Comment:
I think we can directly use `LessThan` here, and we should handle the error when parsing the stack version, as this one can come from user input.
```suggestion
	skipIgnoredFields := append([]string(nil), config.SkipIgnoredFields...)
	stackVersion, err := semver.New(r.stackVersion.Number)
	if err != nil {
		return nil, fmt.Errorf("failed to parse stack version: %w", err)
	}
	if stackVersion.LessThan(semver.MustParse("8.14.0")) {
		// Pre 8.14 Elasticsearch commonly has event.original not mapped correctly, exclude from check: https://github.com/elastic/elasticsearch/pull/106714
		skipIgnoredFields = append(skipIgnoredFields, "event.original")
	}
```